internal/tools/utils.go

```diff
@@ -20,3 +28,79 @@ func Runtime() (string, bool) {
 
 	return "", false
 }
+
+type FailedRequestError int
+
+func (e FailedRequestError) Error() string {
+	return fmt.Sprintf("request failed with status %d", e)
+}
+
+func GetLatestTag(ctx context.Context, owner, repo string) (string, error) {
+	url := fmt.Sprintf("https://api.github.com/repos/%s/%s/releases/latest", owner, repo)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
```

Comment:
have you tried using API https://github.com/google/go-github ?

Reply:
Yes, I played around with the API a bit, but it was a lot of overhead for two `GET` requests. The API would also need to be wrapped because it's a little clunky so I would only use it if we had a lot more interaction with _GitHub_.